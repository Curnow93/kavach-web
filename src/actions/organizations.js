import axios from 'axios';
import {
  ADD_ORGANIZATION,
  ADD_ORGANIZATIONS,
  SET_ORGANIZATIONS_LOADING,
  RESET_ORGANIZATIONS,
  ORGANIZATIONS_API,
  SET_SELECTED_ORGANIZATION,
} from '../constants/organizations';

export const getOrganizations = () => {
  return (dispatch, getState) => {
    dispatch(loadingOrganizations());
    return axios
      .get(ORGANIZATIONS_API + '/my')
      .then((response) => {
        dispatch(addOrganizationsList(response.data));
      })
      .catch((error) => {
        console.log(error.message);
      });
  };
};

export const getOrganization = (id) => {
  return (dispatch, getState) => {
    dispatch(loadingOrganizations());
    return axios
      .get(ORGANIZATIONS_API + '/' + id)
      .then((response) => {
        dispatch(getOrganizationByID(response.data));
        dispatch(stopOrganizationsLoading());
      })
      .catch((error) => {
        console.log(error.message);
      });
  };
};

export const addOrganization = (data) => {
  return (dispatch, getState) => {
    dispatch(loadingOrganizations());
    return axios
      .post(ORGANIZATIONS_API, data)
      .then((response) => {
        dispatch(getOrganizationByID(response.data));
        dispatch(setSelectedOrganization(response.data.id));
        dispatch(stopOrganizationsLoading());
      })
      .catch((error) => {
        console.log(error.message);
      });
  };
};

export const updateOrganization = (data) => {
  return (dispatch, getState) => {
    dispatch(loadingOrganizations());
    return axios
      .put(ORGANIZATIONS_API + '/' + data.id, data)
      .then((response) => {
        dispatch(getOrganizationByID(response.data));
        dispatch(stopOrganizationsLoading());
      })
      .catch((error) => {
        console.log(error.message);
      });
  };
};

export const deleteOrganization = (id) => {
  return (dispatch, getState) => {
    dispatch(loadingOrganizations());
    return axios
      .delete(ORGANIZATIONS_API + '/' + id)
      .then(() => {
        dispatch(resetOrganizations());
      })
      .catch((error) => {
        console.log(error.message);
      });
  };
};

export const setSelectedOrganization = (id) => ({
  type: SET_SELECTED_ORGANIZATION,
  payload: id,
});

const loadingOrganizations = () => ({
  type: SET_ORGANIZATIONS_LOADING,
  payload: true,
});

const stopOrganizationsLoading = () => ({
  type: SET_ORGANIZATIONS_LOADING,
  payload: false,
});

const getOrganizationByID = (data) => ({
  type: ADD_ORGANIZATION,
  payload: data,
});

const addOrganizationsList = (data) => ({
  type: ADD_ORGANIZATIONS,
  payload: data,
});

const resetOrganizations = () => ({
  type: RESET_ORGANIZATIONS,
});
