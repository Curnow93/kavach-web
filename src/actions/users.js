import axios from 'axios';
import { ADD_USERS, SET_USERS_LOADING, RESET_USERS, USERS_API } from '../constants/users';

export const getUsers = () => {
  return (dispatch, getState) => {
    dispatch(loadingUsers());
    return axios
      .get(USERS_API + '/' + getState().organizations.selected + '/users')
      .then((response) => {
        dispatch(addUsersList(response.data));
        dispatch(stopUsersLoading());
      })
      .catch((error) => {
        console.log(error.message);
      });
  };
};

export const addUser = (data) => {
  return (dispatch, getState) => {
    dispatch(loadingUsers());
    return axios
      .post(USERS_API + '/' + getState().organizations.selected + '/users', data)
      .then(() => {
        dispatch(resetUsers());
        dispatch(stopUsersLoading());
      })
      .catch((error) => {
        console.log(error.message);
      });
  };
};

export const deleteUser = (id) => {
  return (dispatch, getState) => {
    dispatch(loadingUsers());
    return axios
      .delete(USERS_API + '/' + getState().organizations.selected + '/users/' + id)
      .then(() => {
        dispatch(resetUsers());
        dispatch(stopUsersLoading());
      })
      .catch((error) => {
        console.log(error.message);
      });
  };
};

const loadingUsers = () => ({
  type: SET_USERS_LOADING,
  payload: true,
});

const stopUsersLoading = () => ({
  type: SET_USERS_LOADING,
  payload: false,
});

const addUsersList = (data) => ({
  type: ADD_USERS,
  payload: data,
});

const resetUsers = () => ({
  type: RESET_USERS,
});
