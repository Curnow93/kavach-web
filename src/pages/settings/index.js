import React from 'react';
import { Button, Form, Input, Space, Divider } from 'antd';
import { useSelector, useDispatch } from 'react-redux';
import {
  getOrganisation,
  updateOrganisation,
  deleteOrganisation,
} from './../../actions/organisations';

function OrganisationEdit() {
  const dispatch = useDispatch();

  const { organisation, selected } = useSelector((state) => {
    return {
      organisation: state.organisations.details[state.organisations.selected],
      selected: state.organisations.selected,
    };
  });

  React.useEffect(() => {
    dispatch(getOrganisation(selected));
  }, [dispatch, selected]);

  return (
    <Space direction="vertical">
      <Form
        name="organisation_edit"
        layout="vertical"
        onFinish={(values) => dispatch(updateOrganisation({ ...organisation, ...values }))}
        initialValues={organisation}
        style={{
          width: '400px',
        }}
      >
        <Form.Item name="title" label="Title">
          <Input placeholder="Title" />
        </Form.Item>
        <Form.Item name="description" label="Description">
          <Input.TextArea placeholder="Description" />
        </Form.Item>
        <Form.Item>
          <Button form="organisation_edit" type="primary" htmlType="submit" block>
            Save
          </Button>
        </Form.Item>
      </Form>
      <Divider style={{ color: 'red' }} orientation="left">
        Danger zone
      </Divider>
      <Button
        danger
        onClick={() =>
          dispatch(deleteOrganisation(organisation.id)).then(window.location.reload(false))
        }
      >
        Default
      </Button>
    </Space>
  );
}

export default OrganisationEdit;
