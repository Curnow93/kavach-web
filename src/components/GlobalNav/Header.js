import React from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { Layout, Divider, Row, Col } from 'antd';
import { MenuUnfoldOutlined, MenuFoldOutlined, LogoutOutlined } from '@ant-design/icons';
import { toggleSider } from '../../actions/settings';

function Header() {
  const { Header: HeaderAnt } = Layout;
  // const [options, setOptions] = React.useState([]);
  const {
    settings: {
      sider: { collapsed },
    },
  } = useSelector((state) => state);
  const dispatch = useDispatch();
  const MenuFoldComponent = collapsed ? MenuUnfoldOutlined : MenuFoldOutlined;

  return (
    <HeaderAnt className="layout-header">
      <Row>
        <Col xs={2} sm={4}>
          <MenuFoldComponent
            style={{ fontSize: '20px' }}
            className="trigger"
            onClick={() => dispatch(toggleSider())}
          />
          <Divider type="vertical" />
        </Col>
        <Col xs={22} sm={20}>
          <a href={process.env.REACT_APP_KRATOS_PUBLIC_URL + '/self-service/browser/flows/logout'}>
            <LogoutOutlined />
            Logout
          </a>
        </Col>
      </Row>
    </HeaderAnt>
  );
}

export default Header;
