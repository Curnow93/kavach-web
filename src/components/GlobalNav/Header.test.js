import React from 'react';
import { useHistory } from 'react-router-dom';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { useDispatch, Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { mount } from 'enzyme';

import '../../matchMedia.mock';
import Header from './Header';
import { MenuUnfoldOutlined, MenuFoldOutlined } from '@ant-design/icons';
import * as actions from '../../actions/settings';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useDispatch: jest.fn(),
}));

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useHistory: jest.fn(),
}));

jest.mock('../../actions/settings', () => ({
  toggleSider: jest.fn(),
}));

describe('Header component', () => {
  let store;
  let mockedDispatch;

  let state = {
    settings: {
      navTheme: 'dark',
      title: 'Dega',
      sider: {
        collapsed: true,
      },
    },
    organisations: {
      ids: [1],
      details: {
        1: { id: 1, title: 'title', description: 'description', permission: { role: 'member' } },
      },
      loading: false,
      selected: 1,
    },
  };
  store = mockStore(() => state);
  store.dispatch = jest.fn(() => ({}));
  mockedDispatch = jest.fn(() => Promise.resolve({}));
  useDispatch.mockReturnValue(mockedDispatch);
  describe('snapshot testing', () => {
    it('should render the component', () => {
      const tree = mount(
        <Provider store={store}>
          <Router>
            <Header />
          </Router>
        </Provider>,
      );
      expect(tree).toMatchSnapshot();
    });
  });
  describe('component testing', () => {
    let wrapper;
    it('should call MenuUnfoldOutlined', (done) => {
      actions.toggleSider.mockReset();
      const push = jest.fn();
      useHistory.mockReturnValueOnce({ push });

      wrapper = mount(
        <Provider store={store}>
          <Router>
            <Header />
          </Router>
        </Provider>,
      );

      wrapper.find(MenuUnfoldOutlined).props().onClick();
      wrapper.update();

      expect(actions.toggleSider).toHaveBeenCalledWith();
      done();
    });
    it('should call MenuFoldOutlined', (done) => {
      actions.toggleSider.mockReset();
      const push = jest.fn();
      useHistory.mockReturnValueOnce({ push });

      state.settings.sider.collapsed = false;
      store = mockStore(() => state);

      wrapper = mount(
        <Provider store={store}>
          <Router>
            <Header />
          </Router>
        </Provider>,
      );

      wrapper.find(MenuFoldOutlined).props().onClick();
      wrapper.update();

      expect(actions.toggleSider).toHaveBeenCalledWith();
      done();
    });
  });
});
