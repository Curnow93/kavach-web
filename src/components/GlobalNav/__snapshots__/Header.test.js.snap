// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component snapshot testing should render the component 1`] = `ReactWrapper {}`;
